api/comments/views.py

```diff
@@ -159,13 +163,24 @@ class CommentDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, Comm
     required_read_scopes = [CoreScopes.NODE_COMMENTS_READ]
     required_write_scopes = [CoreScopes.NODE_COMMENTS_WRITE]
 
-    serializer_class = CommentDetailSerializer
+    serializer_class = NodeCommentDetailSerializer
     view_category = 'comments'
     view_name = 'comment-detail'
 
     # overrides RetrieveAPIView
     def get_object(self):
-        return self.get_comment()
+        comment = self.get_comment()
+
+        if isinstance(comment.target.referent, Node):
+            comment_node = comment.target.referent
+        elif isinstance(comment.target.referent, (NodeWikiPage,
+                                                  StoredFileNode)):
+            comment_node = Guid.load(comment.target.referent.node).referent
+
+        if comment_node.is_registration:
+            self.serializer_class = RegistrationCommentDetailSerializer
```

Comment:
I don't believe I can use get_serializer_class here. This logic is for the call to a particular comment....
e.g. http://localhost:8000/v2/comments/ehar9ypk7wt8/
get_serializer_class is used for listing comments in a node/registration...
e.g. http://localhost:8000/v2/registrations/avn9t/comments/?filter[target]=avn9t